src/fn_call.rs

```diff
@@ -214,11 +216,35 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a + 'mir>: crate::MiriEvalContextExt<'
                 //
                 // `libc::syscall(NR_GETRANDOM, buf.as_mut_ptr(), buf.len(), GRND_NONBLOCK)`
                 // is called if a `HashMap` is created the regular way.
-                match this.read_scalar(args[0])?.to_usize(this)? {
-                    318 | 511 => {
-                        return err!(Unimplemented(
-                            "miri does not support random number generators".to_owned(),
-                        ))
+                match this.read_scalar(args[0])?.to_usize(this)? as i64 {
+                    // SYS_getrandom on x86_64 and x86 respectively
+                    318 | 355 => {
```

Comment:
Can we some make sure we only run with the right syscall number on the right architecture?